src/ErrorBoundary.ts

```diff
@@ -18,6 +18,15 @@ const initialState: ErrorBoundaryState = {
   error: null,
 };
 
+const originalConsoleError = console.error;
+
+window.addEventListener("error", (event) => {
+  if (event.error.isShowBoundary) {
+    event.preventDefault();
+    console.error = () => {};
+  }
+});
```

Comment:
This being in the module scope seems problematic for two reasons:
- It would break server rendering (if `window` is undefined).
- It's a side effect that would linger even if there were no mounted error boundaries. I'm pretty sure we would not want this. If we do something like this, we should likely _only_ do it when there is at least one error boundary that's mounted.

That being said, the approach to override `console.error` does not seem like something we can do. That method could be called for many reasons between when an error is thrown and when a boundary re-renders and commits. Overriding it could cause a lot of developer confusion and even interfere with metrics/error tracking (since this check isn't even wrapped in a DEV conditional).

We could _possibly_ extend the method with something that only intercepts "suppressed" error objects, but even that seems iffy to me. aka something like...
```js
console.error = (...args) => {
  if (args.length === 1) {
    const firstArg = args[0];
    if (
      typeof firstArg === "object" &&
      firstArg != null &&
      firstArg._suppressLogging
    ) {
      return;
    }
  }

  originalConsoleError(...args);
};
```

Reply:
1. Move all window.addEventListener into the lifecycle methods.
2. Delete the console.error override.